SAMtemplates/main_template.yaml

```diff
@@ -0,0 +1,356 @@
+            - "PatientNHSNumber"
+            - "PharmacyODSCode"
+            - "TaskID"
+            - "LineItemID"
+            - "TerminalStatusIndicator"
+            - "LastModified"
+            - "Timestamp"
+            - "RequestMessage"
+            ProjectionType: "INCLUDE"
+          ProvisionedThroughput: 
+            ReadCapacityUnits: "5"
+            WriteCapacityUnits: "5"
+        - IndexName: "PrescriptionIDIndex"
+          KeySchema:
+            - AttributeName: "PrescriptionID"
+              KeyType: "HASH"
+            - AttributeName: "PatientNHSNumber"
+              KeyType: "RANGE"
+          Projection:
+            NonKeyAttributes:
+            - "PharmacyODSCode"
+            - "TaskID"
+            - "LineItemID"
+            - "TerminalStatusIndicator"
+            - "LastModified"
+            - "RequestID"
+            - "Timestamp"
+            - "RequestMessage"
+            ProjectionType: "INCLUDE"
+          ProvisionedThroughput: 
+            ReadCapacityUnits: "5"
+            WriteCapacityUnits: "5"
+        - IndexName: "PatientNHSNumberIndex"
+          KeySchema:
+            - AttributeName: "PatientNHSNumber"
+              KeyType: "HASH"
+            - AttributeName: "PrescriptionID"
+              KeyType: "RANGE"
+          Projection:
+            NonKeyAttributes:
+            - "PharmacyODSCode"
+            - "TaskID"
+            - "LineItemID"
+            - "TerminalStatusIndicator"
+            - "LastModified"
+            - "RequestID"
+            - "Timestamp"
+            - "RequestMessage"
+            ProjectionType: "INCLUDE"
+          ProvisionedThroughput:
+            ReadCapacityUnits: "5"
+            WriteCapacityUnits: "5"
+        - IndexName: "PharmacyODSCodeIndex"
+          KeySchema:
+            - AttributeName: "PharmacyODSCode"
+              KeyType: "HASH"
+            - AttributeName: "PrescriptionID"
+              KeyType: "RANGE"
+          Projection:
+            NonKeyAttributes:
+            - "PatientNHSNumber"
+            - "TaskID"
+            - "LineItemID"
+            - "TerminalStatusIndicator"
+            - "LastModified"
+            - "RequestID"
+            - "Timestamp"
+            - "RequestMessage"
+            ProjectionType: "INCLUDE"
+          ProvisionedThroughput:
+            ReadCapacityUnits: "5"
+            WriteCapacityUnits: "5"
+  DynamoDBWritePolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Roles:
+        - !GetAtt UpdatePrescriptionStatusResources.Outputs.LambdaRoleName
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - dynamodb:BatchWriteItem
+              - dynamodb:DeleteItem
+              - dynamodb:PutItem
+              - dynamodb:UpdateItem
+            Resource:
+              - !Sub "arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${AWS::StackName}-PrescriptionStatusUpdates"
```

Comment:
Can we tweak these to be direct references e.g.:
`- !GetAtt PrescriptionStatusUpdatesTable.Arn`
and
`!Join ["", [!GetAtt PrescriptionStatusUpdatesTable.Arn, "/index/*"]]`

Reply:
Direct references to DynamoDB resources have been added.